examples/esp-todomvc-react/js/model/eventProcessors/TodoListPostEventProcessor.js

```diff
@@ -0,0 +1,49 @@
+/*global app*/
+
+(function () {
+    "use strict";
+
+    app.model.TodoListPostEventProcessor = {
+        process: function (model) {
+            var filter,
+                id,
+                filteredTodoItems = [],
+                incompleteCount = 0,
+                completeCount = 0,
+                count = 0,
+                allChecked = true,
+                todoItem;
+            if (model.footer.filter === app.model.Filter.complete) {
+                filter = function (todoItem) { return todoItem.complete; };
+            }
+            else if (model.footer.filter === app.model.Filter.active) {
+                filter = function (todoItem) { return !todoItem.complete; };
+            } else {
+                filter = function () { return true; };
+            }
+            for (id in model.todoItemsById) {
+                if (model.todoItemsById.hasOwnProperty(id)) {
+                    todoItem = model.todoItemsById[id];
+                    if (filter(todoItem)) {
+                        filteredTodoItems.push(todoItem);
+                        allChecked = todoItem.complete && allChecked;
+                    }
+
+                    if (!todoItem.complete) {
+                        incompleteCount++;
+                    } else {
+                        completeCount++;
+                    }
+
+                    count++;
+                }
+            }
+            model.main.filteredTodoItems = filteredTodoItems;
+            model.main.toggleAll.visible = filteredTodoItems.length > 0;
```

Comment:
Yes, this is deliberate, the model doesn't only contain storage data, it is the model of the entire UI state including things like visibility. Ideally your components/controllers are completely dumb, and can directly data bind off the model without any additional logic. 

Generally you don't want the UI state changing outside the router. Except for minor cases like the editing value inside a focused text box, or a tooltip expanding and collapsing on hover - the model probably doesn't need reflect this.

@sonic78 what do you think?